src/pybind/mgr/dashboard/module.py

```diff
@@ -339,6 +340,7 @@ def serve(self):
         self.shutdown_event.clear()
         NotificationQueue.stop()
         cherrypy.engine.stop()
+        cherrypy.engine.subscribe('stop', __cov.stop)
```

Comment:
Would not be better to subscribe to stop before the engine stops?

Reply:
Hey @alfonsomthd. It would be better to do this right after cherrypy.engine.subscribe('after_request', __cov.save) as calling it here is creating confusion. Thanks for the suggestion.